src/main/java/mate/academy/security/AuthenticationServiceImpl.java

```diff
@@ -6,41 +6,37 @@
 import mate.academy.lib.Inject;
 import mate.academy.lib.Service;
 import mate.academy.model.User;
-import mate.academy.service.ShoppingCartService;
 import mate.academy.service.UserService;
 import mate.academy.util.HashUtil;
 
 @Service
 public class AuthenticationServiceImpl implements AuthenticationService {
     @Inject
     private UserService userService;
-    @Inject
-    private ShoppingCartService shoppingCartService;
 
     @Override
-    public User login(String email, String password) throws AuthenticationException {
-        Optional<User> userFromDb = userService.findByEmail(email);
-        if (userFromDb.isPresent() && matchPasswords(password, userFromDb.get())) {
-            return userFromDb.get();
+    public User register(String email, String password) throws RegistrationException {
+        Optional<User> optionalUserByEmail = userService.findByEmail(email);
+        if (optionalUserByEmail.isPresent()) {
+            throw new RegistrationException("Email " + email + "already registered");
         }
-        throw new AuthenticationException("Incorrect email or password!");
+        User newUser = new User();
+        newUser.setEmail(email);
+        newUser.setPassword(password);
+        return userService.add(newUser);
     }
 
     @Override
-    public User register(String email, String password) throws RegistrationException {
-        if (userService.findByEmail(email).isPresent()) {
-            throw new RegistrationException("User with this email is already registered!");
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> optionalUserByEmail = userService.findByEmail(email);
+        if (optionalUserByEmail.isEmpty()) {
+            throw new AuthenticationException("no user with email:" + email);
+        }
+        User user = optionalUserByEmail.get();
+        if (user.getPassword().equals(HashUtil.hashPassword(password, user.getSalt()))) {
```

Comment:
Ensure that the `HashUtil.hashPassword` method is correctly implemented and that the salt is generated and stored when the user is registered. The current implementation assumes that the salt is already set, which may not be the case.